fl4health/servers/nnunet_server.py

```diff
@@ -212,7 +215,7 @@ def update_before_fit(self, num_rounds: int, timeout: Optional[float]) -> None:
 
             # If config contains nnunet_plans, server side initialization of plans
             # Else client side initialization with nnunet_plans from client
-            if self.fl_config.get("nnunet_plans") is not None:
+            if server_nnunet_plans_exist:
```

Comment:
The code that wraps the on_fit_config function could go within the scope of this if statement. There's no need to update the config functions with the nnunet_plans if we didn't change the config and the nnunet_plans are already in there

Reply:
We talked about this briefly. We'll keep it where it is for now, since it might save someone who forgets the plans in the configs, but did have them in the `fl_config`.